lib/spack/spack/test/cmd/ci.py

```diff
@@ -1404,3 +1404,57 @@ def test_ci_generate_temp_storage_url(tmpdir, mutable_mock_env_path,
                 # Cleanup job should be 2nd to last, just before rebuild-index
                 assert('stage' in cleanup_job)
                 assert(cleanup_job['stage'] == stages[-2])
+
+
+def test_ci_generate_read_broken_specs_url(tmpdir, mutable_mock_env_path,
+                                           env_deactivate, install_mockery,
+                                           mock_packages, monkeypatch):
+    """Verify that `broken-specs-url` works as intended"""
+    spec_a = Spec('a')
+    spec_a.concretize()
+    a_dag_hash = spec_a.dag_hash()
+
+    spec_flattendeps = Spec('flatten-deps')
+    spec_flattendeps.concretize()
+    flattendeps_dag_hash = spec_flattendeps.dag_hash()
+
+    # Mark 'a' as broken (but not 'flatten-deps')
+    broken_spec_a_path = str(tmpdir.join(a_dag_hash))
+    with open(broken_spec_a_path, 'w') as bsf:
+        bsf.write('')
+
+    # Test that `spack ci generate` notices this broken spec and fails.
+    filename = str(tmpdir.join('spack.yaml'))
+    with open(filename, 'w') as f:
+        f.write("""\
+spack:
+  specs:
+    - flatten-deps
+    - a
+  mirrors:
+    some-mirror: https://my.fake.mirror
+  gitlab-ci:
+""")
+        f.write('    broken-specs-url: "file://{0}"\n'.format(tmpdir.strpath))
+        f.write("""\
```

Comment:
I'm sure calling `write` a few times here works fine, but I usually prefer:

```
repl_text = '{0} value'.format('replacement')
"""some long
   string with a {0} as
      well as indentation""".format(repl_text)
```

Reply:
Thanks for the tip! For some reason I got it in my head that str.format() didn't work with triple quotes.